### ISSUE TITLE
wazuh-manager is not starting because of API configuration

### ISSUE THREAD

**Issue by esposig** [wazuh/wazuh-puppet#1178]
Hello,

I just upgraded my wazuh stack to 4.9.2 with puppet and I noticed this: https://github.com/wazuh/wazuh-puppet/blob/v4.9.2/templates/wazuh_api_yml.erb#L5 does not reflect the correct expected type.

```
Dec 02 11:47:49 wazuh env[1342176]: Traceback (most recent call last):
Dec 02 11:47:49 wazuh env[1342176]:   File "/var/ossec/api/scripts/wazuh_apid.py", line 320, in <module>
Dec 02 11:47:49 wazuh env[1342176]:     from api.configuration import read_yaml_config
Dec 02 11:47:49 wazuh env[1342176]:   File "/var/ossec/framework/python/lib/python3.10/site-packages/api/configuration.py", line 343, in <module>
Dec 02 11:47:49 wazuh env[1342176]:     api_conf = read_yaml_config()
Dec 02 11:47:49 wazuh env[1342176]:   File "/var/ossec/framework/python/lib/python3.10/site-packages/api/configuration.py", line 326, in read_yaml_config
Dec 02 11:47:49 wazuh env[1342176]:     configuration = fill_dict(default_conf, configuration, schema)
Dec 02 11:47:49 wazuh env[1342176]:   File "/var/ossec/framework/python/lib/python3.10/site-packages/api/configuration.py", line 183, in fill_dict
Dec 02 11:47:49 wazuh env[1342176]:     raise APIError(2000, details=validation_exc.message) from None
Dec 02 11:47:49 wazuh env[1342176]: api.api_exception.APIError: 2000 - Some parameters are not expected in the configuration file (WAZUH_PATH/api/configuration/api.yaml). Please check the documentation for further details: https://documentation.wazuh.com/4.9/user-manual/api/configuration.html#api-configuration-options: '0.0.0.0' is not of type 'array'.
Dec 02 11:47:49 wazuh env[1342157]: wazuh-apid: Configuration error. Exiting
Dec 02 11:47:49 wazuh systemd[1]: wazuh-manager.service: Control process exited, code=exited, status=1/FAILURE
Dec 02 11:47:49 wazuh systemd[1]: wazuh-manager.service: Failed with result 'exit-code'.
Dec 02 11:47:49 wazuh systemd[1]: Failed to start wazuh-manager.service - Wazuh manager.
```

Can you check that default values for the API use the correct types ?

Thank you.
Guillaume

**Comment by nick-fox** [comment]
This has been fixed in 90baab33eec82d0fed009e621a5a49cec198581d

However, an element of this change needs porting to 4.9.1 and 4.9.2

-      $wazuh_api_host = '0.0.0.0'
+      $wazuh_api_host = ['0.0.0.0']

**Comment by fcaffieri** [comment]
Hello @esposig,
As @nick-fox correctly points out, the issue has been identified and fixed in commit [90baab3](https://github.com/wazuh/wazuh-puppet/commit/90baab33eec82d0fed009e621a5a49cec198581d). The problem was that the API host configuration required an array type, but was being set as a string.
The fix changes the default value in params_manager.pp from:

 ```
$wazuh_api_host = '0.0.0.0'
 ```
 
to:

 ```
$wazuh_api_host = ['0.0.0.0']
 ```

This fix will be included in version 4.10. Until then, you can modify your configuration to use an array format for the host value in versions 4.9.1 and 4.9.2.

**Comment by esposig** [comment]
Hello!

Ok, thank you for your answers both! I'll wait the 4.10 then.

Best.
